examples/pbr/main.js

```diff
@@ -0,0 +1,1411 @@
+                var v = this.computeHammersleyReverse( i );
+                hammersley.push( u );
+                hammersley.push( v );
+            }
+            //console.log( this._hammersley );
+            return hammersley;
+        },
+
+        updateAlbedo: function () {
+            this._albedoTexture = this.createTextureFromColor( this.convertColor( this._config.albedo ), true, this._albedoTexture );
+        },
+
+        updateShaderPBR: function () {
+
+            var nbSamples = this._config.nbSamples;
+            if ( !this._uniformHammersleySequence[ nbSamples ] ) {
+                var sequence = this.computeHammersleySequence( nbSamples );
+                var uniformHammersley = osg.Uniform.createFloat2Array( sequence, 'uHammersleySamples' );
+                this._uniformHammersleySequence[ nbSamples ] = uniformHammersley;
+            }
+            var uniformHammerslay = this._uniformHammersleySequence[ nbSamples ];
+            uniformHammerslay.dirty();
+
+
+            this._shaders.forEach( function ( config ) {
+
+                var stateSet = config.stateSet;
+
+                var shaderConfig = osg.objectMix( {
+                    nbSamples: nbSamples,
+                    environmentType: this._config.environmentType,
+                    pbr: this._config.pbr,
+                    format: this._config.format,
+                    mobile: this._config.mobile
+                }, config.config );
+
+                var program = this.createShaderPBR( shaderConfig );
+
+                stateSet.setAttributeAndModes( program );
+                stateSet.addUniform( uniformHammerslay );
+
+            }.bind( this ) );
+
+        },
+
+        updateEnvironment: function () {
+
+            if ( this._config.environmentType === 'cubemapSeamless' ) {
+                this.setCubemapSeamless();
+            } else {
+                this.setPanorama();
+            }
+
+            this.setBackgroundEnvironment();
+            this.updateEnvironmentRotation();
+            this.updateShaderPBR();
+        },
+
+        convertColor: function ( color ) {
+
+            var r, g, b;
+
+            // rgb [255, 255, 255]
+            if ( color.length === 3 ) {
+                r = color[ 0 ];
+                g = color[ 1 ];
+                b = color[ 2 ];
+
+            } else if ( color.length === 7 ) {
+
+                // hex (24 bits style) '#ffaabb'
+                var intVal = parseInt( color.slice( 1 ), 16 );
+                r = intVal >> 16;
+                g = intVal >> 8 & 0xff;
+                b = intVal & 0xff;
+            }
+
+            var result = [ 0, 0, 0, 1 ];
+            result[ 0 ] = r / 255.0;
+            result[ 1 ] = g / 255.0;
+            result[ 2 ] = b / 255.0;
+            //console.log( result );
+            return result;
+        }
+
+
+    };
+
+    // convert rgbe image to float texture
+    var TextureRGBEToFloatTexture = function ( texture, dest, textureTarget ) {
```

Comment:
main.js starting to get very big... perhaps export more classes in separate js files